webapp/src/pages/Map.tsx

```diff
@@ -269,53 +290,106 @@ function MapPage() {
     }
   }, [])
 
+  const createPointFeature = (point: number[]): GeoJSON.Feature<GeoJSON.Geometry> => {
+    return {
+      type: 'Feature',
+      geometry: {
+        type: 'Point',
+        coordinates: [...point],
+      },
+      properties: {},
+    }
+  }
+
+  const isDateInRange = (date: Date, startDate: Date, endDate: Date): boolean => {
+    return date >= startDate && date <= endDate
+  }
+
+  // Effect for handling polygon updates
   useEffect(() => {
```

Comment:
Not necessary in this PR, but this should be using useMemo instead of useEffect -> useEffect commonly causes re-renders, which can create performance issues